crates/typst-syntax/src/parser.rs

```diff
@@ -506,7 +508,8 @@ fn math_args(p: &mut Parser) {
         match p.current_text() {
             ")" => break,
             ";" => {
-                maybe_wrap_in_math(p, arg, named);
+                maybe_wrap_in_math(p, arg, count, named);
+                count = 0;
```

Comment:
While doing so I noticed something: it looks like `count == 0` implies `namable == true`, except in the case of an unexpected token (although I don't really think there are any unexpected tokens in math). Should we increase count if `p.unexpected()`? If so, should we remove `namable` or leave it for readability?

Reply:
There are unexpected tokens in Math: `$ */ $`.

But I think if it's unexpected it doesn't matter much how it's wrapped, since that only affects evaluation, which won't trigger if there are syntax errors.

I think `namable` makes things a bit more readable, so let's keep it.